TypeScript errors in resources/js/react components break consumer typecheck

## Expected Behavior

`resources/js/react/**` and `resources/js/shared/**` should type-check cleanly against the current published `@artisanpack-ui/react` types, so that consumer apps copying / aliasing these sources into their own builds can run `tsc --noEmit` without errors.

## Current Behavior

Running `tsc --noEmit` in a consumer app that imports the bundled React components surfaces 18 errors across 6 files. They fall into 5 categories:

### 1. `size` prop on `Input` / `Select` (7 errors)

`@artisanpack-ui/react` declares `InputProps extends Omit<InputHTMLAttributes, 'prefix' | 'size'>` and `SelectProps extends Omit<SelectHTMLAttributes, 'size'>` — the `size` prop is intentionally omitted.

The forms React components pass `size="sm"`:

- `react/components/admin/ConditionalLogicEditor.tsx` lines 210, 219, 241, 256, 271, 288
- `react/components/admin/FieldPalette.tsx` line 148
- `react/components/admin/SubmissionsList.tsx` lines 356, 387

Fix: drop `size` or replace with the appropriate `className` per the design system's sizing classes.

### 2. `FieldOption[]` missing index signature (2 errors)

`react/components/fields/ChoiceField.tsx` lines 64 (Select) and 89 (Radio):

```
Type 'FieldOption[]' is not assignable to type 'SelectOption[]' / 'RadioOption[]'.
  Type 'FieldOption' is not assignable ... Index signature for type 'string' is missing in type 'FieldOption'.
```

`SelectOption` / `RadioOption` from `@artisanpack-ui/react` require an index signature. Either widen `FieldOption` or map to the expected shape before passing in.

### 3. `LayoutField.tsx` dynamic tag (3 errors)

`react/components/fields/LayoutField.tsx`:

- Line 21: `Cannot find namespace 'JSX'` — under React 19's new JSX runtime, `JSX.IntrinsicElements` must be imported from `react` (e.g. `keyof React.JSX.IntrinsicElements`) rather than referenced as a global.
- Line 24: derived `Tag` typed as `string | number | symbol` so it can't be used as a JSX component — needs to be narrowed to `keyof JSX.IntrinsicElements` (and the variable should be a capitalized const).

### 4. `ConditionalLogic.action` type incompatibility

`react/components/admin/FieldEditor.tsx` line 244:

```
Type 'GenericConditionalLogic | null' is not assignable to type 'ConditionalLogic | null | undefined'.
  Types of property 'action' are incompatible.
    Type 'string' is not assignable to type 'ConditionalAction'.
```

`GenericConditionalLogic.action` is typed `string`; the consuming `ConditionalLogic` expects the `ConditionalAction` union. Tighten the generic shape's `action` to the union, or convert at the boundary.

### 5. Broken import path / implicit `any` in `shared/`

- `shared/conditionalLogic.ts` line 18 and `shared/validation.ts` line 13:
  ```
  Cannot find module '../../types/artisanpack-forms' or its corresponding type declarations.
  ```
  Path resolves to `resources/types/artisanpack-forms` which doesn't exist in the package — either the file moved or the import path is stale.
- `shared/conditionalLogic.ts` line 309: parameter `rule` implicitly `any`.

## Environment

- `@artisanpack-ui/react` ^1.0.0
- `react` 19
- TypeScript strict mode (`strict: true`)
- artisanpack-ui/forms 1.1.1 (vendored React sources)

## Possible Solution

Triage by category above. Items 1, 3, 5 are mechanical; 2 and 4 require deciding whether to widen the package's own type definitions or convert at the call site.

Happy to send a PR for the mechanical ones if useful.

--- a/resources/js/react/components/admin/ConditionalLogicEditor.tsx
+++ b/resources/js/react/components/admin/ConditionalLogicEditor.tsx
@@ -207,7 +207,7 @@ export function ConditionalLogicEditor( {
 						<Select
 							value={logic.action}
 							onChange={( e ) => handleActionChange( e.target.value )}
-							size="sm"
+							className="select-sm"
 							options={availableActions}
 							optionValue="value"
 							optionLabel="label"
@@ -216,7 +216,7 @@ export function ConditionalLogicEditor( {
 						<Select
 							value={logic.logic}
 							onChange={( e ) => handleLogicTypeChange( e.target.value )}
-							size="sm"
+							className="select-sm"
 							options={[
 								{ value: 'all', label: 'all' },
 								{ value: 'any', label: 'any' },
@@ -238,7 +238,7 @@ export function ConditionalLogicEditor( {
 									<Select
 										value={rule.field}
 										onChange={( e ) => handleUpdateRule( index, { field: e.target.value } )}
-										size="sm"
+										className="select-sm"
 										options={availableFields.map( ( f ) => ( {
 											value: f.uuid,
 											label: f.label || f.name,
@@ -253,7 +253,7 @@ export function ConditionalLogicEditor( {
 										onChange={( e ) => handleUpdateRule( index, {
 											operator: e.target.value as ConditionalOperator,
 										} )}
-										size="sm"
+										className="select-sm"
 										options={OPERATOR_DEFINITIONS.map( ( op ) => ( {
 											value: op.value,
 											label: op.label,
@@ -268,7 +268,7 @@ export function ConditionalLogicEditor( {
 											<Select
 												value={String( rule.value ?? '' )}
 												onChange={( e ) => handleUpdateRule( index, { value: e.target.value } )}
-												size="sm"
+												className="select-sm"
 												options={[
 													{ value: '', label: 'Select...' },
 													...fieldOptions.map( ( opt ) => ( {
@@ -285,7 +285,7 @@ export function ConditionalLogicEditor( {
 												value={String( rule.value ?? '' )}
 												onChange={( e ) => handleUpdateRule( index, { value: e.target.value } )}
 												placeholder="Value"
-												size="sm"
+												className="input-sm"
 											/>
 										)
 									)}

--- a/resources/js/react/components/admin/FieldEditor.tsx
+++ b/resources/js/react/components/admin/FieldEditor.tsx
@@ -16,6 +16,7 @@ import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
 import { Button, Checkbox, Input, Select, Tabs, Textarea } from '@artisanpack-ui/react';
 
 import type {
+	ConditionalLogic,
 	FieldOption,
 	FieldType,
 	FieldWidth,
@@ -241,7 +242,9 @@ export function FieldEditor( {
 				content: (
 					<ConditionalLogicEditor
 						value={field.conditional_logic}
-						onChange={( logic ) => updateField( { conditional_logic: logic } )}
+						onChange={( logic ) => updateField( {
+							conditional_logic: logic as ConditionalLogic | null,
+						} )}
 						fields={allFields}
 						excludeFieldId={field.id}
 					/>

--- a/resources/js/react/components/admin/FieldPalette.tsx
+++ b/resources/js/react/components/admin/FieldPalette.tsx
@@ -145,11 +145,10 @@ export function FieldPalette( {
 			{externalSearch === undefined && (
 				<Input
 					type="search"
-					size="sm"
 					placeholder="Search fields..."
 					value={internalSearch}
 					onChange={( e ) => setInternalSearch( e.target.value )}
-					className="w-full"
+					className="input-sm w-full"
 				/>
 			)}
 

--- a/resources/js/react/components/admin/NotificationEditor.tsx
+++ b/resources/js/react/components/admin/NotificationEditor.tsx
@@ -15,10 +15,10 @@ import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
 import { Alert, Button, Checkbox, Input, Loading, Select, Textarea } from '@artisanpack-ui/react';
 
 import type {
+	ConditionalLogic,
 	Form,
 	FormField,
 	FormNotification,
-	NotificationConditionalLogic,
 	NotificationType,
 	NotificationTypeInfo,
 	StoreNotificationRequest,
@@ -588,7 +588,7 @@ export function NotificationEditor( {
 							<ConditionalLogicEditor
 								value={selectedNotification.conditional_logic}
 								onChange={( logic ) => updateNotification( selectedNotification.id, {
-									conditional_logic: logic as NotificationConditionalLogic | null,
+									conditional_logic: logic as ConditionalLogic | null,
 								} )}
 								fields={fields}
 								actions={[

--- a/resources/js/react/components/admin/NotificationEditor.tsx
+++ b/resources/js/react/components/admin/NotificationEditor.tsx
@@ -18,6 +18,7 @@ import type {
 	Form,
 	FormField,
 	FormNotification,
+	NotificationConditionalLogic,
 	NotificationType,
 	NotificationTypeInfo,
 	StoreNotificationRequest,
@@ -587,7 +588,7 @@ export function NotificationEditor( {
 							<ConditionalLogicEditor
 								value={selectedNotification.conditional_logic}
 								onChange={( logic ) => updateNotification( selectedNotification.id, {
-									conditional_logic: logic,
+									conditional_logic: logic as NotificationConditionalLogic | null,
 								} )}
 								fields={fields}
 								actions={[

--- a/resources/js/react/components/admin/SubmissionsList.tsx
+++ b/resources/js/react/components/admin/SubmissionsList.tsx
@@ -353,8 +353,7 @@ export function SubmissionsList( {
 			<div className="flex flex-wrap items-center gap-3">
 				<Input
 					type="search"
-					size="sm"
-					className="w-64"
+					className="input-sm w-64"
 					placeholder="Search submissions..."
 					value={search}
 					onChange={( e ) => {
@@ -384,7 +383,7 @@ export function SubmissionsList( {
 				</div>
 
 				<Select
-					size="sm"
+					className="select-sm"
 					value={dateRange}
 					onChange={( e ) => {
 						setDateRange( e.target.value as SubmissionDateRange );

--- a/resources/js/react/components/fields/LayoutField.tsx
+++ b/resources/js/react/components/fields/LayoutField.tsx
@@ -9,7 +9,10 @@
  * @since      1.1.0
  */
 
+import type { JSX } from 'react';
 import type { FieldComponentProps } from './types';
+
+type HeadingTag = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
 
 /**
  * Renders a heading element (h1-h6).
@@ -18,7 +21,7 @@ export function HeadingField( { field }: FieldComponentProps ) {
 	const level = field.field_config && 'level' in field.field_config
 		? Number( field.field_config.level )
 		: 2;
-	const Tag = `h${Math.min( Math.max( level, 1 ), 6 )}` as keyof JSX.IntrinsicElements;
+	const Tag = ( `h${Math.min( Math.max( level, 1 ), 6 )}` as HeadingTag ) satisfies keyof JSX.IntrinsicElements;
 
 	return (
 		<Tag className={`font-bold ${field.css_classes ?? ''}`}>

--- a/resources/js/react/components/fields/LayoutField.tsx
+++ b/resources/js/react/components/fields/LayoutField.tsx
@@ -18,10 +18,11 @@ type HeadingTag = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
  * Renders a heading element (h1-h6).
  */
 export function HeadingField( { field }: FieldComponentProps ) {
-	const level = field.field_config && 'level' in field.field_config
+	const rawLevel = field.field_config && 'level' in field.field_config
 		? Number( field.field_config.level )
 		: 2;
-	const Tag = ( `h${Math.min( Math.max( level, 1 ), 6 )}` as HeadingTag ) satisfies keyof JSX.IntrinsicElements;
+	const level = Number.isFinite( rawLevel ) ? Math.min( Math.max( rawLevel, 1 ), 6 ) : 2;
+	const Tag = ( `h${level}` as HeadingTag ) satisfies keyof JSX.IntrinsicElements;
 
 	return (
 		<Tag className={`font-bold ${field.css_classes ?? ''}`}>

--- a/resources/js/shared/conditionalLogic.ts
+++ b/resources/js/shared/conditionalLogic.ts
@@ -15,7 +15,7 @@ import type {
 	ConditionalLogicRule,
 	ConditionalOperator,
 	FormField,
-} from '../../types/artisanpack-forms';
+} from '../types/artisanpack-forms';
 
 /**
  * Builds a mapping of field UUIDs to field names and vice versa.
@@ -306,7 +306,7 @@ export function evaluateFieldVisibility(
 	const action = logic.action ?? 'show';
 	const logicType = logic.logic ?? 'all';
 
-	const results = logic.rules.map( ( rule ) =>
+	const results = logic.rules.map( ( rule: ConditionalLogicRule ) =>
 		evaluateRule( rule, formData, uuidToName ),
 	);
 

--- a/resources/js/shared/validation.ts
+++ b/resources/js/shared/validation.ts
@@ -10,7 +10,7 @@
  * @since      1.1.0
  */
 
-import type { FormField, ValidationRules } from '../../types/artisanpack-forms';
+import type { FormField, ValidationRules } from '../types/artisanpack-forms';
 
 /** Layout field types that do not collect data. */
 const LAYOUT_FIELDS = new Set( ['heading', 'paragraph', 'divider', 'html'] );

--- a/resources/js/types/artisanpack-forms.d.ts
+++ b/resources/js/types/artisanpack-forms.d.ts
@@ -130,6 +130,7 @@ export interface ConditionalLogic {
 export interface FieldOption {
     label: string;
     value: string;
+    [key: string]: string;
 }
 
 /**

--- a/src/FormsServiceProvider.php
+++ b/src/FormsServiceProvider.php
@@ -358,7 +358,7 @@ class FormsServiceProvider extends ServiceProvider
     {
         if ( $this->app->runningInConsole() ) {
             $this->publishes( [
-                __DIR__ . '/../resources/types/artisanpack-forms.d.ts' => resource_path( 'types/artisanpack-forms.d.ts' ),
+                __DIR__ . '/../resources/js/types/artisanpack-forms.d.ts' => resource_path( 'types/artisanpack-forms.d.ts' ),
             ], 'forms-types' );
         }
     }
@@ -377,9 +377,9 @@ class FormsServiceProvider extends ServiceProvider
     {
         if ( $this->app->runningInConsole() ) {
             $this->publishes( [
-                __DIR__ . '/../resources/js/react'                     => resource_path( 'js/vendor/artisanpack-forms/react' ),
-                __DIR__ . '/../resources/js/shared'                    => resource_path( 'js/vendor/artisanpack-forms/shared' ),
-                __DIR__ . '/../resources/types/artisanpack-forms.d.ts' => resource_path( 'js/vendor/artisanpack-forms/types/artisanpack-forms.d.ts' ),
+                __DIR__ . '/../resources/js/react'                        => resource_path( 'js/vendor/artisanpack-forms/react' ),
+                __DIR__ . '/../resources/js/shared'                       => resource_path( 'js/vendor/artisanpack-forms/shared' ),
+                __DIR__ . '/../resources/js/types/artisanpack-forms.d.ts' => resource_path( 'js/vendor/artisanpack-forms/types/artisanpack-forms.d.ts' ),
             ], 'forms-react' );
         }
     }
@@ -398,9 +398,9 @@ class FormsServiceProvider extends ServiceProvider
     {
         if ( $this->app->runningInConsole() ) {
             $this->publishes( [
-                __DIR__ . '/../resources/js/vue'                       => resource_path( 'js/vendor/artisanpack-forms/vue' ),
-                __DIR__ . '/../resources/js/shared'                    => resource_path( 'js/vendor/artisanpack-forms/shared' ),
-                __DIR__ . '/../resources/types/artisanpack-forms.d.ts' => resource_path( 'js/vendor/artisanpack-forms/types/artisanpack-forms.d.ts' ),
+                __DIR__ . '/../resources/js/vue'                          => resource_path( 'js/vendor/artisanpack-forms/vue' ),
+                __DIR__ . '/../resources/js/shared'                       => resource_path( 'js/vendor/artisanpack-forms/shared' ),
+                __DIR__ . '/../resources/js/types/artisanpack-forms.d.ts' => resource_path( 'js/vendor/artisanpack-forms/types/artisanpack-forms.d.ts' ),
             ], 'forms-vue' );
         }
     }

--- a/tests/Feature/TypeDefinitionsTest.php
+++ b/tests/Feature/TypeDefinitionsTest.php
@@ -19,7 +19,7 @@ describe( 'TypeScript Type Definitions', function (): void {
             // Find the source path key that ends with the expected file
             $matchingKeys = array_filter(
                 array_keys( $paths ),
-                fn ( $key ) => str_ends_with( $key, 'resources/types/artisanpack-forms.d.ts' ),
+                fn ( $key ) => str_ends_with( $key, 'resources/js/types/artisanpack-forms.d.ts' ),
             );
 
             expect( $matchingKeys )->not->toBeEmpty();
@@ -31,7 +31,7 @@ describe( 'TypeScript Type Definitions', function (): void {
         } );
 
         it( 'has the source type definitions file present', function (): void {
-            $sourcePath = __DIR__ . '/../../resources/types/artisanpack-forms.d.ts';
+            $sourcePath = __DIR__ . '/../../resources/js/types/artisanpack-forms.d.ts';
 
             expect( file_exists( $sourcePath ) )->toBeTrue();
         } );
@@ -40,7 +40,7 @@ describe( 'TypeScript Type Definitions', function (): void {
     describe( 'type definitions content', function (): void {
         beforeEach( function (): void {
             $this->content = file_get_contents(
-                __DIR__ . '/../../resources/types/artisanpack-forms.d.ts',
+                __DIR__ . '/../../resources/js/types/artisanpack-forms.d.ts',
             );
         } );
 

